storage/modules/balance_storage.go

```diff
@@ -448,7 +432,7 @@ func (b *BalanceStorage) Reconciled(
 		b.pendingReconciliationMutex.Unlock()
 	}
 
-	if err := dbTx.Set(ctx, key, new(big.Int).SetInt64(block.Index).Bytes(), true); err != nil {
+	if err := dbTx.Set(ctx, key, utils.BigIntBytes(new(big.Int).SetInt64(block.Index)), true); err != nil {
```

Comment:
🤮 this is so ugly and inefficient

Reply:
why not 
b := make([]byte, 8)
binary.LittleEndian.PutUint64(b, int64(block.Index))  

does bigint do something special to encode the bytes differently?

Reply:
Probably makes the most sense to just write a `GetInt64` and `StoreInt64` method for some things.

`Amount.Value` would still use `Big.Int`

Reply:
yea maybe typedef Big.Int as Amount or something and then you can write wrappers and do optimizations as you see fit.

func (a *Amount) AsBigInt() big.Int {
	if a == nil {
		return *new(big.Int)
	}
	return big.Int(*a)
}

func (a *Amount) AsInt64() int64 {
	if a == nil {
		return 0
	}
	return (*big.Int)(a).Int64()
}

Reply:
Sorry was prototyping some stuff @sidhujag....I actually typed the response to my own message before github showed me yours.

TL;DR I found a way to fix this with a lot less work...